onnxruntime/python/onnxruntime_inference_collection.py

```diff
@@ -522,3 +522,17 @@ def numpy(self):
         Valid only for OrtValues holding Tensors. Throws for OrtValues holding non-Tensors.
         '''
         return self._ortvalue.numpy()
+
+    def to_dlpack(self):
```

Comment:
Overall, I am wondering if all dlpack related python changes can be confined in orttraining/python folder?

Reply:
This method is within ORTValue. Are you suggesting creating a new class inheriting from it on ort training code?

I believe DLPack can benefit inference code too

Reply:
How about my original suggestion of making DLPack part of Inference?

We just remove all #ifdefs for training and let it live in Inference side. DLPack allows exchange of tensors between many DL frameworks and that can be leveraged for data loaders on inference too, for example. This is not a training specific thing, although training introduced it

It doesnt break Inference and never gets in the execution path if not explicitly used, will be consumed by Training and Inference could leverage it too in the future

Reply:
Removed public APIs